libfreerdp/core/transport.c

```diff
@@ -891,6 +993,52 @@ int transport_write(rdpTransport* transport, wStream* s)
 	}
 
 	LeaveCriticalSection(&(transport->WriteLock));
+fail:
+	if (!length)
+		return writtenlength;
+	return status;
+}
+
+void transport_register_default_io_callbacks(rdpUpdate* update)
+{
+	rdpIoUpdate *io = calloc(1, sizeof(rdpIoUpdate));
```

Comment:
Why allocate here? `rdpIoUpdate io = { 0 };` is most likely the way to go

Reply:
i using ptr inside  update structure, so structure does not copying but just accessed by ptr, without allocation it must be static at least, or it may be copied using memcpy inside implementation i guess, what do you say ?

Reply:
changed  as requested.